src/main/java/it/eng/idsa/dataapp/util/MessageUtil.java

```diff
@@ -77,4 +163,48 @@ private String createContractAgreement(Path dataLakeDirectory) {
 		}
 		return contractAgreement;
 	}
+	
+	private Connector getSelfDescription() {
+		URI eccURI = null;
+		try {
+			eccURI = new URI(eccProperties.getRESTprotocol(), null, eccProperties.getHost(), eccProperties.getRESTport(), null, null, null);
+		} catch (URISyntaxException e) {
+			logger.error("Could not create URI for Self Description request.", e);
+		}
+
+		try {
+			return new Serializer().deserialize(restTemplate.getForObject(eccURI, String.class), Connector.class);
+		} catch (RestClientException e) {
+			logger.error("Could not fetch self description from ECC", e);
+			return null;
+		} catch (IOException e) {
+			logger.error("Could not deserialize self description to Connector instance", e);
+			return null;
+		}
+	}
+	
+	private String getSelfDescriptionAsString() {
```

Comment:
Can this one be replaced with MultipartMessageProcessor.serializeToJsonLD?

Reply:
I can change the return value to MultipartMessageProcessor.serializeToJsonLD(getSelfDescription()), but I would still suggest having it in a private method just to skip making many try/catch blocks and unnecessarily bloating the code.

Reply:
Created new ticket, to address this serialize throws exception